File: frontend/src/services/levelService.js
import axiosClient from "../api/axiosClient";

const levelService = {
  getAll() {
    return axiosClient.get("/v1/levels");
  },

  getById(id) {
    return axiosClient.get(`/v1/levels/${id}`);
  },

  create(data) {
    return axiosClient.post("/v1/levels", data);
  },

  update(id, data) {
    return axiosClient.put(`/v1/levels/${id}`, data);
  },

  lock(id) {
    return axiosClient.post(`/v1/levels/${id}/lock`);
  },

  unlock(id) {
    return axiosClient.post(`/v1/levels/${id}/unlock`);
  },
};

export default levelService;

File: backend/src/main/java/com/englishlearning/backend/service/LevelService.java
package com.englishlearning.backend.service;

import com.englishlearning.backend.dto.request.LevelCreateRequest;
import com.englishlearning.backend.dto.request.LevelUpdateRequest;
import com.englishlearning.backend.dto.response.LevelResponse;

import java.util.List;

public interface LevelService {

    List<LevelResponse> getAll();

    LevelResponse getById(Long id);

    LevelResponse create(LevelCreateRequest request);

    LevelResponse update(Long id, LevelUpdateRequest request);


    void lock(Long id);

    void unlock(Long id);
}